src/components/People/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+
+import { PeopleFilters } from '../PeopleFilters';
+import { Loader } from '../../Loader';
+import { PeopleTable } from '../PeopleTable';
+import { Person } from '../../../types';
+import { getPeople } from '../../../api';
+import { filterPeople, sortPeople } from '../../../helpers';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { slug = '' } = useParams();
+
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  useEffect(() => {
+    const getPeopleFromServer = async () => {
+      setIsLoading(true);
+      setHasError(false);
+
+      try {
+        const peopleFromServer = await getPeople();
+
+        setPeople(peopleFromServer);
+      } catch {
+        setHasError(true);
+      }
+
+      setIsLoading(false);
+    };
+
+    getPeopleFromServer();
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    return filterPeople(people, sex, query, centuries);
+  }, [people, sex, query, centuries]);
+
+  const sortedPeople = useMemo(() => {
+    return sortPeople(filteredPeople, sort, order);
+  }, [filteredPeople, sort, order]);
```

Comment:
I want to praise you here for splitting logic. It's really great for future code maintenance and debugging.